packages/kbn-search-connectors/lib/generate_connector_name.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+
+import { toAlphanumeric } from '../utils/to_alphanumeric';
+import { indexOrAliasExists } from './exists_index';
+
+const GENERATE_INDEX_NAME_ERROR = 'generate_index_name_error';
+
+export const generateConnectorName = async (
+  client: ElasticsearchClient,
+  connectorType: string,
+  userConnectorName?: string
+): Promise<{ connectorName: string; indexName: string }> => {
+  const prefix = toAlphanumeric(connectorType);
+  if (!prefix || prefix.length === 0) {
+    throw new Error('Connector type or connectorName is required');
+  }
+
+  if (userConnectorName) {
+    let indexName = `connector-${userConnectorName}`;
```

Comment:
LGTM, one small nit, agentless connectors would need to have indexName prefixed with `content-` can we add a flag to this function that would control this prefix? e.g. if connector is maanged set the prefix to `content`

Reply:
That makes sense. Let's merge https://github.com/elastic/kibana/pull/202179 first so I can use the const there without merge conflicts.

Reply:
Ahh it's reusing the same logic! Good point

Reply:
@jedrazb added a native flag that will prepend the index name with `content-` (although in practice this will do nothing right now, but good to have)